CMake: MCUboot: fix image hex address in RAM Load mode

JarmouniA opened the pull request:
In MCUBoot RAM LOAD mode, the app slot addr is taken as the hex-addr passed to imgtool to create slot0 & 1 images, which is only correct if the grand-parent node's (usually the flash node) absolute address is actually 0, because the addresses of all the flash fixed-partitions in-tree are relative to their grand-parent node's address.

Hence, use the newly added `dt_fixed_partition_addr()` CMake function to get the absolute address of app slots partitions.
`dt_fixed_partition_addr()` is equivalent to the MACRO
https://github.com/zephyrproject-rtos/zephyr/blob/8a72d776c5af6522478155adc91a5f4edb2794e3/include/zephyr/devicetree/fixed-partitions.h#L133

erwango reviewed (changes requested):
Probably a good thing but nack on the board impact part for now

JarmouniA committed:
cmake: modules: extensions: introduce dt_fixed_partition_addr function

Introduce a new dt_fixed_partition_addr() function that returns the
absolute address of a fixed partition, meaning the sum of the base
address of the containing grand-parent node and the partition's offset.

Signed-off-by: Abderrahmane JARMOUNI <git@jarmouni.me>

JarmouniA committed:
cmake: mcuboot: use dt_fixed_partition_addr() to get slot addr

In RAM LOAD mode, the app slot addr is taken as the hex-addr passed to
imgtool, which is only correct if the grand-parent node's (usually the
flash node) absolute address is actually 0, because the addresses of all
the flash fixed-partitions in-tree are relative to their grand-parent
node's address.

Hence, use the newly added dt_fixed_partition_addr() function to get the
absolute address of app slots partitions.

Signed-off-by: Abderrahmane JARMOUNI <git@jarmouni.me>

nordicjm commented on cmake/mcuboot.cmake:
```diff
@@ -137,9 +137,9 @@ function(zephyr_mcuboot_tasks)
+    dt_fixed_partition_addr(slot0_partition_address PATH ${slot0_partition})
     dt_nodelabel(slot1_partition NODELABEL "slot1_partition" REQUIRED)
-    dt_reg_addr(slot1_partition_address PATH ${slot1_partition})
+    dt_fixed_partition_addr(slot1_partition_address PATH ${slot1_partition})
```
I present a complete failure showcase for this:
```
diff --git a/dts/vendor/nordic/nrf52840_partition.dtsi b/dts/vendor/nordic/nrf52840_partition.dtsi
index 1fa90a3ed71..fd51c151379 100644
--- a/dts/vendor/nordic/nrf52840_partition.dtsi
+++ b/dts/vendor/nordic/nrf52840_partition.dtsi
@@ -23,15 +23,24 @@
 			reg = <0x00000000 0x0000C000>;
 		};
 
-		slot0_partition: partition@c000 {
+                slot_partitions: partition@c000 {
+                        compatible = "fixed-subpartitions";
+                        label = "image-0";
+                        reg = <0x0000c000 0xec000>;
+                        ranges = <0x0 0xc000 0xec000>;
+                        #address-cells = <1>;
+                        #size-cells = <1>;
+
+		slot0_partition: partition@0 {
 			label = "image-0";
-			reg = <0x0000C000 0x00076000>;
+			reg = <0x00000000 0x00076000>;
 		};
 
-		slot1_partition: partition@82000 {
+		slot1_partition: partition@76000 {
 			label = "image-1";
-			reg = <0x00082000 0x00076000>;
+			reg = <0x00076000 0x00076000>;
 		};
+};
 
 		/*
 		 * The flash starting at 0x000f8000 and ending at
```
Build `samples/subsys/mgmt/mcumgr/smp_svr` for `nrf52840dk/nrf52840` with `-DFILE_SUFFIX="ram_load"` and observe the huge error. The issue on your board has the correct solution described on https://github.com/zephyrproject-rtos/zephyr/pull/97413#issuecomment-3396095904 you are not setting ranges and then declaring a child node has address 0, **this means it starts at physical address 0x0**, your DTS file is wrong

JarmouniA replied:
```
+      ranges = <0x0 0x70000000 0x1000000>;
 
 	partitions { 
```
Ranges here has no effect on the individual partitions absolute address (seen in devicetree_generated.h), because I think the immediate childnode is `partitions`, not the the `partition@x` itself. Adding `ranges = <.... ` inside `partitions` node gives build errors.
